SwiftCompilerSources/Sources/Basic/SourceLoc.swift

```diff
@@ -0,0 +1,73 @@
+//===--- SourceLoc.swift - SourceLoc bridiging utilities ------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+public struct SourceLoc {
+  public var pointer: UnsafePointer<UInt8>
+
+  public init?(pointer: UnsafePointer<UInt8>?) {
+    guard let pointer = pointer else {
+      return nil
+    }
+    self.pointer = pointer
+  }
+
+  public init?(pointer: UnsafePointer<CChar>?) {
+    guard let pointer = pointer else {
+      return nil
+    }
+    self.pointer = UnsafeRawPointer(pointer).assumingMemoryBound(to: UInt8.self)
+  }
+
+  public init?(bridged: BridgedSourceLoc) {
+    guard let pointer = bridged.pointer else {
+      return nil
+    }
+    self.init(pointer: pointer)
+  }
+
+  public var bridged: BridgedSourceLoc {
```

Comment:
same here

Reply:
This file is in `Basic`.
`var bridged` is used in `DiagnosticEngine.diagnose()` in `AST` to pass the bridged values to `DiagnosticEngine_diagnose()` C function.
I can make this underscored, but it'll be inconsistent with StringRef or ArrayRef's `withBridgedXXX`